File: 704.binary-search.py
#
# @lc app=leetcode id=704 lang=python3
#
# [704] Binary Search
#

# @lc code=start
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        middle = 0
        right = len(nums)
        left = 0
        if target < nums[0] or target > nums[-1]:
            return -1
        while (left < right):
            middle = (left+right) // 2
            if nums[middle] < target:
                left = middle + 1
            elif nums[middle] > target:
                right = middle
            else:
                return middle
        return -1
# @lc code=end

File: 704.binary-search.js
/*
 * @lc app=leetcode id=704 lang=javascript
 *
 * [704] Binary Search
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var search = function(nums, target) {
    let left = 0;
    let right = nums.length;
    while (left < right){
        mid = Math.floor(left + (right-left)/2);
        if (nums[mid] > target){
            right = mid;
        }
        else if (nums[mid] < target){
            left = mid+1;
        }
        else return mid;
    }
    return -1;
};
// @lc code=end

